zookeeper-api/src/main/java/org/apache/helix/zookeeper/api/api/client/RealmAwareZkClient.java

```diff
@@ -305,159 +314,12 @@ public int hashCode() {
-    // For client to init the connection
-    private long _connectInitTimeout = HelixZkClient.DEFAULT_CONNECTION_TIMEOUT;
-
-    // Data access configs
-    private long _operationRetryTimeout = HelixZkClient.DEFAULT_OPERATION_TIMEOUT;
-
-    // Others
-    private PathBasedZkSerializer _zkSerializer;
-
-    // Monitoring
-    private String _monitorType;
-    private String _monitorKey;
-    private String _monitorInstanceName = null;
-    private boolean _monitorRootPathOnly = true;
-
-    public HelixZkClient.ZkClientConfig setZkSerializer(PathBasedZkSerializer zkSerializer) {
-      this._zkSerializer = zkSerializer;
-      return this;
-    }
-
-    public HelixZkClient.ZkClientConfig setZkSerializer(ZkSerializer zkSerializer) {
-      this._zkSerializer = new BasicZkSerializer(zkSerializer);
-      return this;
-    }
-
-    /**
-     * Used as part of the MBean ObjectName. This item is required for enabling monitoring.
-     *
-     * @param monitorType
-     */
-    public HelixZkClient.ZkClientConfig setMonitorType(String monitorType) {
-      this._monitorType = monitorType;
-      return this;
-    }
-
-    /**
-     * Used as part of the MBean ObjectName. This item is required for enabling monitoring.
-     *
-     * @param monitorKey
-     */
-    public HelixZkClient.ZkClientConfig setMonitorKey(String monitorKey) {
-      this._monitorKey = monitorKey;
-      return this;
-    }
-
-    /**
-     * Used as part of the MBean ObjectName. This item is optional.
-     *
-     * @param instanceName
-     */
-    public HelixZkClient.ZkClientConfig setMonitorInstanceName(String instanceName) {
-      this._monitorInstanceName = instanceName;
-      return this;
-    }
-
-    public HelixZkClient.ZkClientConfig setMonitorRootPathOnly(Boolean monitorRootPathOnly) {
-      this._monitorRootPathOnly = monitorRootPathOnly;
-      return this;
-    }
-
-    public HelixZkClient.ZkClientConfig setOperationRetryTimeout(Long operationRetryTimeout) {
-      this._operationRetryTimeout = operationRetryTimeout;
-      return this;
-    }
-
-    public HelixZkClient.ZkClientConfig setConnectInitTimeout(long _connectInitTimeout) {
-      this._connectInitTimeout = _connectInitTimeout;
-      return this;
-    }
-
-    public PathBasedZkSerializer getZkSerializer() {
-      if (_zkSerializer == null) {
-        _zkSerializer = new BasicZkSerializer(new SerializableSerializer());
-      }
-      return _zkSerializer;
-    }
-
-    public long getOperationRetryTimeout() {
-      return _operationRetryTimeout;
-    }
-
-    public String getMonitorType() {
-      return _monitorType;
-    }
-
-    public String getMonitorKey() {
-      return _monitorKey;
-    }
-
-    public String getMonitorInstanceName() {
-      return _monitorInstanceName;
-    }
-
-    public boolean isMonitorRootPathOnly() {
-      return _monitorRootPathOnly;
-    }
-
-    public long getConnectInitTimeout() {
-      return _connectInitTimeout;
+  class RealmAwareZkClientConfig {
```

Comment:
Shall we change it ConnectionConfig? Conceptually this is the same as HelixZkClient.ZkConnectionConfig. In HelixZkClient the ClientConfig is about the monitor key, serializer etc. Let us keep the naming to concept consistent, otherwise, it can be confusing.